tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,458 @@
+        return serialize(
+            Value(
+                val,
+                AccessMode.STRING,
+                access_mode=AccessMode.READ_ONLY,
+                allowed_values=allowed,
+            )
+        )
+
+    def _set_state(self, state: State):
+        # LOGGER.info(f"Transitioned State: [{self.state} -> {state}]")
+        self.status.state = state
+
+
+class EigerRESTAdapter(
+    HTTPAdapter, EigerStreamAdapter, EigerMonitorAdapter, EigerFileWriterAdapter
+):
+    """An Eiger adapter which parses the commands sent to the HTTP server."""
+
+    device: EigerDevice
+
+    # TODO: Make API version setable in the config params?
+    @HTTPEndpoint.get(f"/{DETECTOR_API}" + "/config/{parameter_name}")
+    async def get_config(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for requesting configuration variables from the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the given parameter.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        param = request.match_info["parameter_name"]
+
+        if hasattr(self.device.settings, param):
+            attr = self.device.settings[param]
+
+            data = serialize(
+                Value(
+                    attr["value"],
+                    attr["metadata"]["value_type"].value,
+                    access_mode=(
+                        attr["metadata"]["access_mode"].value
+                        if hasattr(attr["metadata"], "access_mode")
+                        else AccessMode.READ_ONLY.value
+                    ),
+                )
+            )
+        else:
+            data = serialize(Value("None", "string", access_mode="None"))
+
+        return web.json_response(data)
+
+    @HTTPEndpoint.put(
+        f"/{DETECTOR_API}" + "/config/{parameter_name}", include_json=True
+    )
+    async def put_config(self, request: web.Request) -> web.Response:
+        """A HTTP Endpoint for setting configuration variables for the Eiger.
+
+        Args:
+            request (web.Request): The request object that takes the given parameter
+            and value.
+
+        Returns:
+            web.Response: The response object returned given the result of the HTTP
+                request.
+        """
+        param = request.match_info["parameter_name"]
+
+        response = await request.json()
+
+        if self.device.get_state()["value"] != State.IDLE.value:
+            LOGGER.warning("Eiger not initialized or is currently running.")
+            return web.json_response(serialize(SequenceComplete(7)))
+        elif (
+            hasattr(self.device.settings, param)
+            and self.device.get_state()["value"] == State.IDLE.value
+        ):
+            attr = response["value"]
+
+            LOGGER.debug(f"Changing to {attr} for {param}")
+
+            setattr(self.device.settings, param, attr)
```

Comment:
Should we be allowing the setting of arbitrary parameters? I would think we should check if `settings` contains `param`, and only then perform an assignment. This would technically allow someone to redefine a method

Reply:
this is already done a few lines above?

Reply:
would this be better then? 
```python
elif (
            param in vars(self.device.settings).keys()
            and self.device.get_state()["value"] == State.IDLE.value
        ):
            attr = response["value"]
            setattr(self.device.settings, param, attr)
            return web.json_response(serialize(SequenceComplete(8)))```

Reply:
avoids using `hasattr` too

Reply:
I reckon the "nice" way to do this is to implement a `__setattr__` in the settings class, much like you currently have a `__getattr__`

Reply:
i made a `__getitem__` method, so in this case would it also be `__setitem__`?

Reply:
Yes

Reply:
So turns out making a `__setitem__` overwrites the property setter function for element....